hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializer.java

```diff
@@ -109,6 +109,12 @@ public void initializeInstance(RowProcessingState rowProcessingState) {
 			return;
 		}
 
+		final EntityInitializer parentEntityInitializer = parentAccess.asEntityInitializer();
```

Comment:
Shouldn't this rather be the following to also fix possible issues when the association is part of a component?

```suggestion
		final EntityInitializer parentEntityInitializer = parentAccess.findFirstEntityDescriptorAccess().asEntityInitializer();
```

Reply:
done